From 437f6b3d59f2102f5974bfc430176725f2ed67c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 25 Sep 2019 16:34:43 +0200
Subject: [PATCH 41/43] renderer-native: Actually use shadow fb when using
 software rendering

The commit 'renderer/native: Use shadow fb on software GL if preferred'
attempted to force using a shadow fb when using llvmpipe in order to
speed up blending, but instead only did so when llvmpipe AND the drm
device explicityl asked for it.

Now instead always force it for llvmpipe and other software rendering
backends, and otherwise just query the drm device (i.e.
DRM_CAP_DUMB_PREFER_SHADOW).

https://gitlab.gnome.org/GNOME/mutter/merge_requests/807
---
 src/backends/native/meta-renderer-native.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backends/native/meta-renderer-native.c b/src/backends/native/meta-renderer-native.c
index 5c01375c8..21d00502c 100644
--- a/src/backends/native/meta-renderer-native.c
+++ b/src/backends/native/meta-renderer-native.c
@@ -3213,11 +3213,11 @@ should_force_shadow_fb (MetaRendererNative *renderer_native,
     case COGL_GPU_INFO_ARCHITECTURE_SANDYBRIDGE:
     case COGL_GPU_INFO_ARCHITECTURE_SGX:
     case COGL_GPU_INFO_ARCHITECTURE_MALI:
-      return FALSE;
+      break;
     case COGL_GPU_INFO_ARCHITECTURE_LLVMPIPE:
     case COGL_GPU_INFO_ARCHITECTURE_SOFTPIPE:
     case COGL_GPU_INFO_ARCHITECTURE_SWRAST:
-      break;
+      return TRUE;
     }
 
   kms_fd = meta_gpu_kms_get_fd (primary_gpu);
-- 
2.23.0

